src/component/variantPage/SideBar.tsx

```diff
@@ -75,8 +91,34 @@ class SideBar extends React.Component<SideBarProps>
     }
 
     @action.bound
-    onSearch () {
-        this.props.history.push(`/variant/${this.inputText}`);
+    async onSearch() {
+        if (isVariantValid(`${this.inputText}`).isValid) {
+            let hasResult = false;
+            // check if the variant has response
+            await client.fetchVariantAnnotationSummaryGET({variant: this.inputText!}).then(function(response){
```

Comment:
Ideally, you shouldn't use both `await` and `.then` together. Try to do it this way:

```suggestion
            const response = await client.fetchVariantAnnotationSummaryGET({variant: this.inputText!});
```